script.js

```diff
@@ -0,0 +1,100 @@
+//Variables
+let todos = []
+
+//Element storage
+$addBtn = $('#new-task')
+$newTaskInput = $('#new-task-input')
+$addTask = $('#add-task')
+$taskList = $('#todo-list')
+$linkList = $('#link-list')
+
+//Click listeners
+$addBtn.on('click', showForm)
+$addTask.on('click', addTask)
+
+
+
+//on page load, get list of todos from localstorage
+$(document).ready(function(){
+    todos = JSON.parse(localStorage.getItem('todos'))
+    if (!todos) todos = [];
+    renderList()
+
+    chrome.bookmarks.getTree(function(bookmarkTreeNodes) {
+        updateLinkList(bookmarkTreeNodes)
+    });
+})
+
+
+//Functions
+//Show add todo button and input
+function showForm(){
+    $newTaskInput.toggleClass('hidden')
+    $addTask.toggleClass('hidden')
+    $addBtn.toggleClass('hidden')
+}
+
+//add todo to list and save to local storage
+function addTask(){
+    //update localstorage
+    todos.push({
+        item: $newTaskInput.val(),
+        done: false
+    })
+    localStorage.setItem('todos', JSON.stringify(todos))
+    //add to ui
+    renderList()
+    $newTaskInput.val('')
+    $newTaskInput.toggleClass('hidden')
+    $addTask.toggleClass('hidden')
+    $addBtn.toggleClass('hidden')
+}
+
+//toggle todo state (done or not done)
+function toggleTodo(num) {
+    return (() => {
+        todos[num].done = !todos[num].done
```

Comment:
is there a reason for iife with return. seems like callback could be done without it